ssv/runner_signatures.go

```diff
@@ -43,30 +43,49 @@ func (b *BaseRunner) validatePartialSigMsgForSlot(
 		return errors.New("invalid partial sig slot")
 	}
 
-	for _, msg := range psigMsgs.Messages {
+	// Get signer
+	msgSigner := psigMsgs.Messages[0].Signer // signer is the same in all psigMsgs.Messages and len(psigMsgs.Messages) > 0 (guaranteed by psigMsgs.Validate())
 
-		// Check if knows it has the validator index share
-		validatorShare, ok := b.Share[msg.ValidatorIndex]
-		if !ok {
-			return errors.New("unknown validator index")
-		}
+	// Get committee (unique for runner)
+	var shareSample *types.Share
+	for _, share := range b.Share {
+		shareSample = share
+		break
+	}
+	if shareSample == nil {
+		return errors.New("can not get committee because there is no share in runner")
+	}
+	committee := shareSample.Committee
 
-		// Check if signer is in committee
-		signerInCommittee := false
-		for _, operator := range validatorShare.Committee {
-			if operator.Signer == msg.Signer {
-				signerInCommittee = true
-				break
-			}
-		}
-		if !signerInCommittee {
-			return errors.New("unknown signer")
+	// Check if signer is in committee
+	signerInCommittee := false
+	for _, operator := range committee {
+		if operator.Signer == msgSigner {
+			signerInCommittee = true
+			break
 		}
 	}
+	if !signerInCommittee {
+		return errors.New("unknown signer")
+	}
```

Comment:
according to the flow this would stop processing... won't this defeat our purpose?

I think I said you can leave it only if it won't stop the processing flow

Reply:
even though I see tests are passing so not sure what's going on

Reply:
This checks if the operator is on the committee. This should be correct independently, no? Or are we thinking of committee changes in terms of operators?

Reply:
yeah you are right